.buildscript/deploy_snapshot.sh

```diff
@@ -6,7 +6,7 @@
 # http://benlimmer.com/2013/12/26/automatically-publish-javadoc-to-gh-pages-with-travis-ci/
 
 SLUG="square/javapoet"
-JDK="oraclejdk8"
+JDK="oraclejdk7"
```

Comment:
I think we need to build on JDK7 regardless, due to the strange binary-compatibility stuff that @tbroyer mentioned: just the presence of JDK8 signatures during compilation can potentially cause the JDK7 .jar to be malformed.